lib/galaxy/webapps/galaxy/api/biotools.py

```diff
@@ -0,0 +1,299 @@
+        'mirror': [],
+        'canonicalID': '',
+        'tag': [],
+        'elixirInfo': {
+            'elixirStatus': '',
+            'elixirNode': ''
+        },
+        'language': [],
+        'license': '',
+        'cost': '',
+        'docs': {
+            'docsHome': '',
+            'docsTermsOfUse': '',
+            'docsDownload': '',
+            'docsCitationInstructions': ''
+        },
+        'credits': {
+            'creditsDeveloper': [],
+            'creditsContributor': [],
+            'creditsInstitution': [],
+            'creditsInfrastructure': [],
+            'creditsFunding': []
+        },
+        'contact': [{
+            'contactEmail': "conf.contactEmail",
+            'contactURL': '',
+            'contactName': "conf.contactName",
+            'contactTel': '',
+            'contactRole': []
+        }]
+    }
+    return gen_dict
+
+
+def build_function_dict(json_tool):
+    """
+    Extract information from a galaxy json tool and return a list of functions in the json biotools format
+    :param json_tool: galaxy json tool
+    :return: list of functions in the json biotools format
+    :rtype: list
+    """
+    list_func = list()
+    listinps = inputs_extract(json_tool['inputs'])
+    listoutps = ouputs_extract(json_tool['outputs'])
+    func_dict = {
+        'functionDescription': json_tool['description'],
+        'functionName': [{
+            'uri': "http://edamontology.org/operation_0004",
+            'term': 'EDAM label placeholder'
+        }],
+        'output': listoutps,
+        'input': listinps,
+        'functionHandle': " "
+    }
+    list_func.append(func_dict)
+    return list_func
+
+
+def inputs_extract(inputs_json):
+    """
+    Extract type data param of a galaxy json tool inputs and return a list of dictionary in the json biotools format
+    :param inputs_json: inputs part of a json tool
+    :return: list of dictionary in the json biotools format
+    :rtype: list
+    """
+
+    def inputs_extract_data(data_json):
+        """
+        Save param type data from a json tool galaxy in a list
+        :param data_json:
+        :return: None
+        """
+        list_format = list()
+        for edam_format in data_json['edam']['edam_formats']:
+            list_format.append({'uri': edam_to_uri(edam_format), 'term': 'EDAM label placeholder'})
+
+        if len(set(data_json['edam']['edam_data'])) == 1:
+            listdata.append({'dataType': {'uri': edam_to_uri(data_json['edam']['edam_data'][0]),
+                                          'term': 'EDAM label placeholder'},
+                             'dataFormat': list_format,
+                             'dataHandle': ", ".join(data_json['extensions']),
+                             'dataDescription': data_json['name']
+                             })
+        else:
+            listdata.append({'dataType': {'uri': 'WARNING, SEVERAL EDAM DATA', 'term': 'EDAM label placeholder'},
```

Comment:
This is supposed to deal with multiple edam formats for a tool, which is possible, so this should be reworked.